### PR TITLE
WIP:add scheduler to Clean Zombie Consumers

### LINKED ISSUE #116: should create a peridic job to clean zoombie consumer
```
**:6379> xinfo consumers company  cg1
1) 1) "name"
   2) "c1"
   3) "pending"
   4) (integer) 3
   5) "idle"
   6) (integer) 4785470
```
once the idle is longer than a duration(6 hours maybe), we should del this consumer。

Once a consumer was killed unexpectedly, that consumer will be left as an Orphan.

### PR DESCRIPTION
**Pull request by lilien1010**
close https://github.com/vmihailenco/taskq/issues/116

### CODE REVIEW

**Comment by lilien1010** [comment]
still need more test, but you can review them first.

File: redisq/queue.go [hunk 1]
```diff
@@ -329,6 +336,25 @@ func (q *Queue) scheduleDelayed(ctx context.Context) (int, error) {
 	return len(bodies), nil
 }
 
+func (q *Queue) scheduleCleanZombieConsumers(ctx context.Context) (int, error) {
```
Comment (vmihailenco) [inline redisq/queue.go]: Just cleanZombieConsumers?

File: redisq/queue.go [hunk 2]
```diff
@@ -329,6 +336,25 @@ func (q *Queue) scheduleDelayed(ctx context.Context) (int, error) {
...
+	consumers, err := q.redis.XInfoConsumers(ctx, q.stream, q.streamGroup).Result()
+
+	if err != nil {
+		return 0, err
+	}
+
+	for _, consumer := range consumers {
+		if consumer.Name == q.streamConsumer {
```
Comment (vmihailenco) [inline redisq/queue.go:348]: Would be nice to add a comment for future readers why you skip this.

File: queue.go [hunk 3]
```diff
@@ -107,6 +110,10 @@ func (opt *QueueOptions) Init() {
 		opt.WaitTimeout = 10 * time.Second
 	}
 
+	if opt.ZombieConsumerTimeout == 0 {
+		opt.ZombieConsumerTimeout = 6 * time.Hour
+	}
```
Comment (vmihailenco) [inline queue.go:116]: What about `ConsumerIdleTimeout`? Looks cleaner to me.

**Comment by vmihailenco** [comment]
Looks good overall :+1:

File: queue.go (hunk 3 above)
Comment (lilien1010) [reply queue.go:116]: done

File: redisq/queue.go (hunk 2 above)
Comment (lilien1010) [reply redisq/queue.go:348]: done

File: redisq/queue.go (hunk 1 above)
Comment (lilien1010) [reply redisq/queue.go]: done

File: redisq/queue.go (hunk 2 above)
Comment (vmihailenco) [reply redisq/queue.go:348]: I guess it is not really needed and is just an optimization? Just checking :)

**Review by vmihailenco — APPROVED** [review]

File: redisq/queue.go (hunk 2 above)
Comment (lilien1010) [reply redisq/queue.go:348]: adding this line , because I am not sure what will happen if we delete the current consumer, just in case.

**Pull request merged**